api/processors/group.py

```diff
@@ -0,0 +1,93 @@
+import datetime
+
+from django.db import transaction
+from django.utils import timezone
+
+from api.models import CoordinatorLogEvent, Group, GroupLogEvent, StudentLogEvent, TeacherLogEvent
+from api.models.choices.log_event_types import (
+    CoordinatorLogEventType,
+    GroupLogEventType,
+    StudentLogEventType,
+    TeacherLogEventType,
+)
+from api.models.choices.statuses import (
+    CoordinatorStatus,
+    GroupStatus,
+    StudentStatus,
+    TeacherStatus,
+)
+from api.models.people import CoordinatorManager, TeacherManager
+from api.processors.base import Processor
+
+
+class GroupProcessor(Processor):
+    @classmethod
+    @transaction.atomic
+    def start(cls, group: Group) -> None:
+        cls._create_log_events_start(group)
+        cls._set_statuses_start(group)
+
+    @classmethod
+    def _create_log_events_start(cls, group: Group) -> None:
+        GroupLogEvent.objects.create(group=group, type=GroupLogEventType.STARTED)
+
+        CoordinatorLogEvent.objects.bulk_create(
+            CoordinatorLogEvent(
+                coordinator=coordinator,
+                group=group,
+                type=CoordinatorLogEventType.TOOK_NEW_GROUP,
+            )
+            for coordinator in group.coordinators.iterator()
+        )
+
+        StudentLogEvent.objects.bulk_create(
+            StudentLogEvent(student=student, to_group=group, type=StudentLogEventType.STUDY_START)
+            for student in group.students.iterator()
+        )
+
+        TeacherLogEvent.objects.bulk_create(
+            TeacherLogEvent(teacher=teacher, to_group=group, type=TeacherLogEventType.STUDY_START)
+            for teacher in group.teachers.iterator()
+        )
+
+    @classmethod
+    def _set_statuses_start(cls, group: Group) -> None:
+        timestamp = timezone.now()
+
+        cls._set_status(obj=group, status=GroupStatus.WORKING, status_since=timestamp)
+
+        for category in ("coordinators", "students", "teachers"):
```

Comment:
Are you sure it's more readable than

```
set_coordinators_status_start(group=group, timestamp=timestamp)
set_teachers_status_start(group=group, timestamp=timestamp)
set_students_status_start(group=group, timestamp=timestamp)
```

Reply:
Also, we still have these timestamps passed to every call. Do we really want all of them to be same?

Reply:
> set_coordinators_status_start(group=group, timestamp=timestamp)
> set_teachers_status_start(group=group, timestamp=timestamp)
> set_students_status_start(group=group, timestamp=timestamp)

Fixed